hackrf.h ### Example TX code utilizing the transfer and flush callbacks.  Needed to change line to void flush_callback(void * transfer, int success) {

### What would you like to know?

**Compiling the Example TX code utilizing the transfer and flush callback that is in  in hackrf.h had compile error**
(2023 host code)

copied to another file rfgen_tx.c

Needed to change line
**void flush_callback(hackrf_transfer *transfer) {**
to 
**void flush_callback(void * transfer, int success) {**

To get the source to compile and run.

Otherwise would get the following error 
gcc -o rfgen_tx rfgen_tx.c -lm -lhackrf

```
rfgen_tx.c: In function ‘main’:
rfgen_tx.c:80:36: warning: passing argument 2 of ‘hackrf_enable_tx_flush’ from incompatible pointer type [-Wincompatible-pointer-types]
80 | hackrf_enable_tx_flush(device, flush_callback, NULL);
| ^~~~~~~~~~~~~~
| |
| void (*)(hackrf_transfer ) {aka void ()(struct )}
In file included from rfgen_tx.c:3:
/usr/local/include/libhackrf/hackrf.h:1150:21: note: expected ‘hackrf_flush_cb_fn’ {aka ‘void ()(void , int)’} but argument is of type ‘void ()(hackrf_transfer )’ {aka ‘void ()(struct *)’}
1150 | hackrf_flush_cb_fn callback,

```

Curious if others may have this issue or if I have something in my env that requires the line change?

Verified the signal transmitted and debug printed

hackrf_open
hackrf_open_setup
libusb_get_configuration result: 0
hackrf_open_setup: set_hackrf_configuration result: 0
libusb_claim_interface result: 0
hackrf_set_freq
hackrf_set_txvga_gain
hackrf_enable_tx_flush
hackrf_close


Also Thank-you for the Code Example provided by :
// Transmit a 440Hz triangle wave through FM (144.5MHz) using the libhackrf API
// Copyright (c) 2022 László Baráth "Uncle Dino" HA7DN https://github.com/Sasszem

--- a/host/libhackrf/src/hackrf.h
+++ b/host/libhackrf/src/hackrf.h
@@ -363,7 +363,7 @@ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSI
  *     return 0;
  * }
  * 
- * void flush_callback(hackrf_transfer *transfer) {
+ * void flush_callback(void *ctx, int success) {
  *     pthread_mutex_lock(&mutex);
  *     pthread_cond_broadcast(&cond);
  *     pthread_mutex_unlock(&mutex);

